### PR TITLE
Add program options

### PR DESCRIPTION
**Pull request by KharchenkoMaks**
Added start options for:

- server

1. --host arg, set server host, default: 0.0.0.
2. --port arg, set server listening port, default: 8586
3. --debug, set logging level to debug

- client

1. --log_console, toggle logging to console
2. --debug, set logging level to debug
3. --ip, set remote server address

### CODE REVIEW

File: src/client/client.cpp [hunk 1]
```diff
@@ -14,15 +14,52 @@
...
+    std::string ip = "localhost:8586";
 
-    std::string target_str = "localhost:8586";
+    boost::program_options::options_description desc("Allowed options");
+
+    desc.add_options()
+            ("help", "prints help message")
+            ("log_console", boost::program_options::bool_switch(&show_logs_in_console), "show logs in console")
+            ("debug", boost::program_options::bool_switch(&severity_debug), "enable debug logs")
+            ("ip", boost::program_options::value(&ip), "server remote host, default: localhost:8586");
```
Comment (vladimir-moiseiev) [inline src/client/client.cpp]: 8586 is not a part of ip nor host

File: src/client/client.cpp [hunk 2]
```diff
@@ -14,15 +14,52 @@
...
+    boost::program_options::store(parse_command_line(argc, argv, desc), vm);
+    boost::program_options::notify(vm);
+
+    if (vm.count("help")) {
+        std::cout << desc << "\n";
+        return "";
+    }
+
+    if (severity_debug)
+        logs_init(boost::log::trivial::severity_level::debug, show_logs_in_console);
```
Comment (vladimir-moiseiev) [inline src/client/client.cpp]: severity_debug ? boost::log::trivial::severity_level::debug : boost::log::trivial::severity_level::info

File: src/client/client.cpp [hunk 3]
```diff
@@ -14,15 +14,52 @@
...
+        logs_init(boost::log::trivial::severity_level::info, show_logs_in_console);
+
+    return ip;
+}
+
+int main(int argc, char* argv[]) {
+    std::string target_str = parse_options(argc, argv);
+
+    if (target_str.empty())
+        return 1;
```
Comment (vladimir-moiseiev) [inline src/client/client.cpp]: No logs?!

File: src/server/server.cpp [hunk 4]
```diff
@@ -9,20 +9,53 @@
...
+    boost::program_options::store(parse_command_line(argc, argv, desc), vm);
+    boost::program_options::notify(vm);
+
+    if (vm.count("help")) {
+        std::cout << desc << "\n";
+        return "";
+    }
+
+    if (severity_debug)
+        logs_init(boost::log::trivial::severity_level::debug);
```
Comment (vladimir-moiseiev) [inline src/server/server.cpp]: I saw this code already...
Why is it repeating?

**Review by vladimir-moiseiev — CHANGES_REQUESTED** [review]

File: src/client/client.cpp (hunk 1 above)
Comment (KharchenkoMaks) [reply src/client/client.cpp]: separated to host and port options

File: src/client/client.cpp (hunk 2 above)
Comment (KharchenkoMaks) [reply src/client/client.cpp]: fixed

File: src/client/client.cpp (hunk 3 above)
Comment (KharchenkoMaks) [reply src/client/client.cpp]: the idea was, that of --help parameter passed, print help string and end the program, fixed

File: src/server/server.cpp (hunk 4 above)
Comment (KharchenkoMaks) [reply src/server/server.cpp]: moved parameter parsing to separate function

**Pull request merged**